### PR TITLE
Update OpenMS recipies for 3.0.0 release

### PR DESCRIPTION
**Pull request by timosachsenberg**
Describe your pull request here

Contains the new logic that covers nightly and release builds.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/pyopenms/meta.yaml [hunk 1]
```diff
@@ -1,16 +1,23 @@
...
+{% if 'dev' in version %}
+source:
+  git_url: https://github.com/OpenMS/OpenMS.git
+  git_rev: develop
+  git_depth: 1 # (Defaults to -1/not shallow)
+{% else %}
 source:
   url: https://abibuilder.cs.uni-tuebingen.de/archive/openms/OpenMSInstaller/release/{{ version }}/OpenMS-{{ version }}-src.tar.gz
-  sha256: 2d186f35fe42573a19283605ccfb491ca6da585880fa3725a013975df26bc008
+  sha256: 2b7ee0f4d8f42eb3c17384347df2dba4
```
Comment (timosachsenberg) [inline recipes/pyopenms/meta.yaml:15]:
```suggestion
  md5: 2b7ee0f4d8f42eb3c17384347df2dba4
```

**Comment by timosachsenberg** [comment]
@BiocondaBot please add label

**Comment by timosachsenberg** [comment]
@BiocondaBot please fetch artifacts

**Comment by bgruening** [comment]
Cool. Version 3.0. I have converted it into Draft PR until you are using a proper release. Hope that is ok.

**Comment by timosachsenberg** [comment]
please merge I think this should be ready. Also I think this needs to be merged before I can open a PR for the pyopenms 3.0.0 recipe.

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
So you want to keep this dev if/else ... I see. Lets ship it!

**Pull request merged**